src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentHelper.cs

```diff
@@ -237,7 +249,23 @@ private ViewComponentDescriptor SelectComponent(Type componentType)
                     Resources.FormatViewComponent_IViewComponentFactory_ReturnedNull(descriptor.Type.FullName));
             }
 
+            WriteTelemetryIfEnabled("Microsoft.AspNet.Mvc.BeforeViewComponent", context);
```

Comment:
There are two code paths here, `InvokeCore` and `InvokeCoreAsync` - it depends on whether the calling code is async or not. `Contextualize` is something different entirely and isn't related.